src/silverlabnwb/header.py

```diff
@@ -50,6 +51,8 @@ def from_file(cls, filename):
                     key, value = int(float(words[0])), float(words[1])  # TODO cast later in code
                     fields.append([section, key, value])
                     parsed_fields[section][key] = value
+                else:
+                    warnings.warn("Unrecognised non-blank line in {}: {}".format(filename, line))
```

Comment:
Nice! It would be great to have a test for this too, on a file that contains unrecognised lines. You can check that a warning is raised with [`pytest.warns`](https://docs.pytest.org/en/latest/warnings.html#warns).